code/perception/src/lidar_distance.py

```
@@ -18,19 +18,159 @@
+        """
+        Filtert LiDAR-Punktwolken, führt Clustering durch und veröffentlicht die kombinierten Cluster.
 
-        :param data: a PointCloud2
+        :param data: LiDAR-Punktwolken im ROS PointCloud2-Format.
         """
+
+        # Punktwolken filtern, um irrelevante Daten zu entfernen
         coordinates = ros_numpy.point_cloud2.pointcloud2_to_array(data)
+        filtered_coordinates = coordinates[
+            ~(
+                (-2 <= coordinates["x"])
```

Ruff (0.8.0)

Yoda condition detected

Rewrite as `coordinates["x"] >= -2`

(SIM300)